### PR TITLE
Rewrite ChainerX connection tests

### PR DESCRIPTION
**Pull request by niboshi**
Part of #6401

### CODE REVIEW

**Comment by hvy** [comment]
Basically LGTM but am wondering why the class names are written in `snake_case` 🤔

File: tests/chainerx_tests/op_utils.py [hunk 1]
```diff
@@ -9,7 +9,51 @@
 import chainerx
 
 
-class _OpTest(chainer.testing.function.FunctionTestBase):
+class OpTest(chainer.testing.function.FunctionTestBase):
```
Comment (hvy) [inline tests/chainerx_tests/op_utils.py]: Could you explain why the underscore was removed?

**Comment by hvy** [comment]
jenkins, test this please.

**Comment by hvy** [comment]
Ci failure seems unrelated. LGTM!

**Comment by hvy** [comment]
For the record: Dot may cut the graph during forward propagation depending on input shape (ndim == 0). This means that backward may result in gradients of `None`. Since this backward computation is the "forward computation" in the double backward test. We need to allow `None` outputs from the forward operation when testing double backward.

**Comment by niboshi** [comment]
Added comment.

**Comment by hvy** [comment]
jenkins, test this please.

**Comment by hvy** [comment]
Hm, is (one of the) Jenkins failure related? 🤔

**Comment by niboshi** [comment]
@hvy 
It looks it's just a stochastic numeric failure.

**Comment by niboshi** [comment]
I confirmed this test fails even in master in about the same probability.

**Comment by hvy** [comment]
jenkins, test this please.

**Comment by hvy** [comment]
jenkins, test this please.

**Comment by hvy** [comment]
Jenkins, test this please.

**Comment by hvy** [comment]
LGTM!

**Pull request merged**